### PR TITLE
Refactor resize.c into separate functions.

### PR DESCRIPTION
**Pull request by parazyd**
* **What does this PR do?**

Splits up mutt_resize_screen() into its separate function to improve readability.

* **What are the relevant issue numbers?**

https://github.com/neomutt/neomutt/issues/1149

### CODE REVIEW

File: resize.c [hunk 1]
```diff
@@ -42,50 +43,95 @@
 #endif
 #endif
 
-/**
- * mutt_resize_screen - Called after receiving SIGWINCH
- */
-void mutt_resize_screen(void)
+int mutt_resize_getrows(int rows)
```
Comment (flatcap) [inline resize.c]: We don't need the `rows` parameter.
The old (invalid) value we're passed is just overwritten anyway.

File: resize.c [hunk 2]
```diff
@@ -42,50 +43,95 @@
...
 {
   const char *cp = NULL;
-  int fd;
-  struct winsize w;
-#if !defined(USE_SLANG_CURSES)
-  int SLtt_Screen_Rows, SLtt_Screen_Cols;
-#endif
+  cp = mutt_str_getenv("LINES");
```
Comment (flatcap) [inline resize.c]: These two lines can be combined

File: resize.c [hunk 3]
```diff
@@ -42,50 +43,95 @@
...
+  if (cp && (mutt_str_atoi(cp, &rows) < 0))
+    return 24;
+  return -1;
```
Comment (flatcap) [inline resize.c]: The default of `-1` **is** the original behaviour,
but I think returning a sensible value (24) is a better idea.

File: resize.c [hunk 4]
```diff
@@ -42,50 +43,95 @@
...
+int mutt_resize_getrows(int rows)
 {
   const char *cp = NULL;
-  int fd;
-  struct winsize w;
-#if !defined(USE_SLANG_CURSES)
-  int SLtt_Screen_Rows, SLtt_Screen_Cols;
-#endif
+  cp = mutt_str_getenv("LINES");
+  if (cp && (mutt_str_atoi(cp, &rows) < 0))
```
Comment (flatcap) [inline resize.c]: `atoi` saves the value into `rows` (a local copy of a variable), but it's not returned to the caller.

File: resize.c [hunk 5]
```diff
@@ -42,50 +43,95 @@
...
+    return 24;
+  return -1;
+}
 
-  SLtt_Screen_Rows = -1;
-  SLtt_Screen_Cols = -1;
+int mutt_resize_getcols(int cols)
```
Comment (flatcap) [inline resize.c]: Same comments as for `getrows()`

File: resize.c [hunk 6]
```diff
@@ -42,50 +43,95 @@
...
+{
+  const char *cp = NULL;
+  cp = mutt_str_getenv("COLUMNS");
+  if (cp && (mutt_str_atoi(cp, &cols) < 0))
+    return 80;
+  return -1;
+}
+
+int mutt_resize_getwinsize(winsize w)
```
Comment (flatcap) [inline resize.c]: You're missing a `struct`

File: resize.c [hunk 7]
```diff
@@ -42,50 +43,95 @@
...
+{
+  int fd;
```
Comment (flatcap) [inline resize.c]: We can combine this with the assign, below

File: resize.c (hunk 6 above)
Comment (flatcap) [reply resize.c]: This needs to be a pointer, otherwise we're working with a local copy of the struct

File: resize.c [hunk 8]
```diff
@@ -42,50 +43,95 @@
...
-    cp = mutt_str_getenv("LINES");
-    if (cp && (mutt_str_atoi(cp, &SLtt_Screen_Rows) < 0))
-      SLtt_Screen_Rows = 24;
+  return ret;
+}
+
+#if defined(USE_SLANG_CURSES)
+void mutt_resize_screen_slang(void)
+{
+  struct winsize w;
```
Comment (flatcap) [inline resize.c]: Making this `w = { 0 }` will mean the members are initialised

File: resize.c [hunk 9]
```diff
@@ -42,50 +43,95 @@
...
   stdscr = newwin(0, 0, 0, 0);
   keypad(stdscr, true);
+  mutt_window_reflow();
+}
 #else
-  resizeterm(SLtt_Screen_Rows, SLtt_Screen_Cols);
-#endif
+void mutt_resize_screen_ncurses(void)
+{
+  struct winsize w;
```
Comment (flatcap) [inline resize.c]: `w = { 0 }`

File: resize.c [hunk 10]
```diff
@@ -42,50 +43,95 @@
...
+
+  resizeterm(screenrows, screencols);
   mutt_window_reflow();
 }
+#endif
+
+/**
+ * mutt_resize_screen - Called after receiving SIGWINCH
+ */
+void mutt_resize_screen(void)
```
Comment (flatcap) [inline resize.c]: I don't think we really need this extra level of abstraction.
We can simply rename the two functions to `mutt_resize_screen()` and `#ifdef` them.

**Comment by parazyd** [comment]
I've updated my branch with a little different way of patching it. I think this is better.

**Comment by flatcap** [comment]
I've just made a small tweak (which saves a few lines).
Unless you've anything to add, I think we've spent more than enough time on this :-)

**Comment by parazyd** [comment]
Looks alright. Sorry for all the confusion :D

**Pull request merged**